make CapacityIndicator work with other values of splits, not only splits:10

schilken opened the pull request:
Currently the CapacityIndicator only works with the default value of 10 for splits.

I changed CapacityIndicator to work with other values and IndicatorsPage now uses splits:20.

```
macos_ui git:(dev) sh ./pr_prelaunch_tasks.sh
Formatted 115 files (0 changed) in 0.45 seconds.
Run dart fix --dry-run? [y/n]
y
Computing fixes in macos_ui (dry run)... 10.5s
Nothing to fix!
Run dart fix --apply? [y/n]
y
Computing fixes in macos_ui... 10.4s
Nothing to fix!
No changes to commit
Run tests? [y/n]
y
00:05 +15: /Users/aschilken/flutterdev/forks/macos_ui/test/buttons/popup_button_test.dart: MacosPopupButton tests debugFillProperties
[itemHeight: 24.0, noAutofocus, popupColor: null, menuMaxHeight: null, alignment: AlignmentDirectional.centerStart]
00:18 +119: All tests passed!
```

schilken committed:
make CapacityIndicator work with other values of splits, not only with splits:10

GroovinChip commented:
@whiplashoo is Capacity Indicator something still used by macOS devs? I can't find a reference to it in the HIG.

GroovinChip commented:
@whiplashoo Pinging you again for your thoughts on this

schilken commented:
@GroovinChip  Hi, sorry for not responding. Surely I use it, and I found this:  
https://developer.apple.com/design/human-interface-guidelines/components/status/level-indicators/

GroovinChip reviewed (approved):
Looks good. Please add a test that accounts for varying amounts of steps, and update the version & the change log.

schilken committed:
add unit test for CapacityIndicator with splits 20

schilken committed:
add bugfix line to CHANGELOG

schilken commented:
@GroovinChip  I added an entry in CHANGELOG and a unit test for splits= 20.
I don't know what you mean by "update the version".

I think the unit tests of the CapacityIndicator don't prove anything useful concerning the splits, they just check the properties, not the calculation of the segments, but I have no idea who to test that.

On the IndicatorsPage of the example I changed the splits to 20 to show that the bug is fixed.

GroovinChip commented:
I'll take a look at the tests when I get a moment.

By "update the version" I'm talking about incrementing the version number in the package's `pubspec.yaml`.

schilken committed:
Merge branch 'GroovinChip:dev' into dev

schilken committed:
update version to 1.7.7, move change in CHANGELOG.md to version 1.7.7

GroovinChip reviewed (dismissed):
OK @schilken I was able to review the test you wrote. I think you should take a look at how Flutter tests the `ProgressIndicator` widget. Check it out [here](https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/progress_indicator_test.dart) and add some similar tests. I don't think you need to test it as deeply as they do, but it should give you some ideas and direction.

schilken committed:
add test to check the number of filled segments of discrete CapacityIndicator

schilken committed:
fix warnings in copied sources mock_canvas.dart and recording_canvas.dart

schilken commented:
Hi @GroovinChip, I added two tests to check the number of segments. Thanks for pointing out the flutter tests. That was very helpful; I did not know the canvas_mock yet.

schilken committed:
Merge branch 'dev' into dev

GroovinChip commented:
@schilken looks like you need to format `capacity_indicators_test.dart`, `mock_canvas.dart`, and `recording_canvas.dart`

schilken committed:
format new and changed files

schilken committed:
Merge branch 'dev' of https://github.com/schilken/macos_ui into dev
merge github branch into local

GroovinChip commented:
@schilken Can you pull in the latest from `dev` so I can re-review, please?

schilken committed:
Merge remote-tracking branch 'upstream/dev' into dev

# Conflicts:
#	CHANGELOG.md

schilken commented:
After merging the upstream dev branch into my local branch, I get an exception when clicking the CapacityIndicator on the IndicatorsPage of the example app: 

```
Exception has occurred.
_AssertionError ('package:macos_ui/src/indicators/slider.dart': Failed assertion: line 40 pos 16: 'value >= min && value <= max': is not true.)
```

The ranges 0-100 of CapacityIndicator and 0.0 to 1.0 of Slider are not compatible.

A further test reveals, that the release build just works fine, it's just the connection of CapacityIndicator and slider that triggers the assertError.

GroovinChip commented:
> After merging the upstream dev branch into my local branch, I get an exception when clicking the CapacityIndicator on the IndicatorsPage of the example app: 
> 
> 
> 
> ```
> 
> Exception has occurred.
> 
> _AssertionError ('package:macos_ui/src/indicators/slider.dart': Failed assertion: line 40 pos 16: 'value >= min && value <= max': is not true.)
> 
> ```
> 
> 
> 
> The ranges 0-100 of CapacityIndicator and 0.0 to 1.0 of Slider are not compatible.
> 
> 
> 
> A further test reveals, that the release build just works fine, it's just the connection of CapacityIndicator and slider that triggers the assertError.

It's just a problem in the example app, I address it in #342.

GroovinChip committed:
Merge branch 'dev' into dev

GroovinChip commented on CHANGELOG.md:
```diff
@@ -20,6 +20,8 @@ Other changes
 * Update `MacosScrollBar.thumbVisibility` with the latest change introduced in Flutter 3.7
 * Update `README.md` to address issues [#325](https://github.com/GroovinChip/macos_ui/issues/325) & [#332](https://github.com/GroovinChip/macos_ui/issues/332)
 
+* Fixed a bug where `CapacityIndicator` did only work correctly for splits = 10
```
```suggestion
* Fixed a bug where `CapacityIndicator` only worked correctly for splits = 10
```

GroovinChip reviewed (dismissed):
Just a minor nit in the changelog, & please run flutter pub get to ensure the lockfiles stay updated. We'll be good to go after that 👍

Edit: also please address the failing check

schilken committed:
Update CHANGELOG.md

Co-authored-by: Reuben Turner <groovinchip@gmail.com>

schilken committed:
Merge remote-tracking branch 'upstream/dev' into dev
merge upstream

schilken committed:
Merge remote-tracking branch 'origin/dev' into dev
merge dev

schilken committed:
format indicators_page.dart, run flutter pub get to update lock files

schilken committed:
set dart < 3.0.0

schilken committed:
revert change of spec checksums in Podfile.lock

GroovinChip reviewed (approved):
LGTM 🎉
